packages/transformers/less/src/LessTransformer.js

```diff
@@ -0,0 +1,103 @@
+// @flow strict-local
+
+import {Transformer} from '@parcel/plugin';
+import localRequire from '@parcel/local-require';
+
+// E.g: ~library/file.less
+const WEBPACK_ALIAS_RE = /^~[^/]/;
+
+export default new Transformer({
+  async getConfig({asset, resolve}) {
+    let config = await asset.getConfig(['.lessrc', '.lessrc.js'], {
+      packageKey: 'less'
+    });
+
+    if (config === null) {
+      config = {};
+    }
+
+    config.filename = asset.filePath;
+    config.plugins = [
+      ...(config.plugins || []),
+      urlPlugin({asset}),
+      resolvePathPlugin({asset, resolve})
+    ];
+
+    return config;
+  },
+
+  async transform({asset, config}) {
+    const less = await localRequire('less', asset.filePath);
+    const code = await asset.getCode();
+    let css;
+    try {
+      css = (await less.render(code, config)).css;
+    } catch (err) {
+      // For the error reporter
+      err.fileName = err.filename;
+      err.loc = {
+        line: err.line,
+        column: err.column
+      };
+      throw err;
+    }
+
+    asset.type = 'css';
+    asset.setCode(css);
+    asset.meta.hasDependencies = false;
+    return [asset];
+  }
+});
+
+function urlPlugin({asset}) {
+  return {
+    install(less, pluginManager) {
+      const visitor = new less.visitors.Visitor({
+        visitUrl(node) {
+          node.value.value = asset.addURLDependency(
+            node.value.value,
+            node.currentFileInfo.filename
+          );
+          return node;
+        }
+      });
+
+      visitor.run = visitor.visit;
+      pluginManager.addVisitor(visitor);
+    }
+  };
+}
+
+function resolvePathPlugin({asset, resolve}) {
+  return {
+    install(less, pluginManager) {
+      class LessFileManager extends less.FileManager {
+        supports() {
+          return true;
+        }
+
+        supportsSync() {
+          return false;
+        }
+
+        async loadFile(rawFilename, ...args) {
+          let filename = rawFilename;
+
+          if (WEBPACK_ALIAS_RE.test(filename)) {
+            let correctPath = filename.replace(/^~/, '~/node_modules/');
```

Comment:
Not sure if this should be `~/node_modules/some-module/something.less` or just `some-module/something.less`.
We should support both, we might wanna recommend the easier and less Parcel specific one although not sure as it might clash with relative paths. We could also mention both but that seems a bit much?

Reply:
I don't think that case would be common, assuming someone has `normalize.css` in their `/styles` folder, and then does `@import "normalize.css"` in their `index.less` file, I believe the user will expect their possibly-custom `normalize.css` file to be imported, not the one installed from npm.